editor/scene/canvas_item_editor_plugin.cpp

```diff
@@ -1094,6 +1106,52 @@ void CanvasItemEditor::_on_grid_menu_id_pressed(int p_id) {
 	viewport->queue_redraw();
 }
 
+void CanvasItemEditor::_reset_transform(TransformType p_type) {
+	List<Node *> selection = editor_selection->get_full_selected_node_list();
+	if (selection.is_empty()) {
+		return;
+	}
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	undo_redo->create_action(TTR("Reset Transform"));
+	for (Node *node : selection) {
+		Node2D *res_node = Object::cast_to<Node2D>(node);
+		if (res_node) {
+			switch (p_type) {
+				case TransformType::POSITION:
+					undo_redo->add_undo_method(res_node, "set_position", res_node->get_position());
+					undo_redo->add_do_method(res_node, "set_position", Vector2());
+					break;
+				case TransformType::ROTATION:
+					undo_redo->add_undo_method(res_node, "set_rotation", res_node->get_rotation());
+					undo_redo->add_do_method(res_node, "set_rotation", 0);
+					break;
+				case TransformType::SCALE:
+					undo_redo->add_undo_method(res_node, "set_scale", res_node->get_scale());
+					undo_redo->add_do_method(res_node, "set_scale", Size2(1, 1));
+					break;
+			}
```

Comment:
You could add a continue here. If it's Node2D, it won't be a Control.

```suggestion
			}
			continue;
```